src/process-hooks.ts

```diff
@@ -28,15 +28,25 @@ export default (tagFunction: () => TramOneElement) => {
 	const existingEffects = getEffectStore(TRAM_EFFECT_STORE);
 	const queuedEffects = getEffectStore(TRAM_EFFECT_QUEUE);
 
-	// get all new keys
-	const newKeys = getKeyQueue(TRAM_KEY_QUEUE);
+	// pull new effects that have yet to be processed from the tag
+	// these can appear when a component re-exposes another component at its root
+	const existingNewEffects = tagResult[TRAM_TAG_NEW_EFFECTS] || [];
 
-	// store new effects in the node we just built
+	// store new effects (and the existing new effects) in the node we just built
 	const newEffects = Object.keys(queuedEffects).filter((effect) => !(effect in existingEffects));
-	tagResult[TRAM_TAG_NEW_EFFECTS] = newEffects.map((newEffectKey) => queuedEffects[newEffectKey]);
+	const newEffectFunctions = newEffects.map((newEffectKey) => queuedEffects[newEffectKey]);
+	const existingNewAndBrandNewEffects = existingNewEffects.concat(newEffectFunctions);
```

Comment:
we should write a test to verify that both the top-level and child components' effects are processed